packages/plugin-file/src/index.ts

```
@@ -0,0 +1,116 @@
+import createDebug from 'debug';
+import globby from 'globby';
+import ms from 'ms';
+
+export interface FilePluginOptions {
+  files: string[];
+  reloadInterval?: string | number;
+}
+
+const debug = createDebug('config-dug:plugin:file');
+
+class FilePlugin extends BaseConfigDugPlugin<FilePluginOptions> {
+  private configDugOptions: ConfigDugOptions | undefined;
+  private valueOrigins: Record<string, string[]> = {};
+
+  constructor(options: FilePluginOptions) {
+    super(options);
+    this.pluginOptions = options;
+  }
+
+  public override initialize = async (configDugOptions: ConfigDugOptions): Promise<void> => {
+    this.configDugOptions = configDugOptions;
+    this.initialized = true;
+  };
+
+  public load = async (): Promise<ConfigDugPluginOutput> => {
+    if (!this.initialized) {
+      throw new Error('Plugin not initialized');
+    }
+
+    if (this.configDugOptions === undefined) {
+      throw new Error('ConfigDugOptions not set');
+    }
+
+    this.valueOrigins = {};
+    let values: Record<string, unknown> = {};
+    let nextReloadIn: number | undefined;
+
+    const paths = await globby(this.pluginOptions.files);
+
+    debug('paths', paths);
+
+    for (const file of paths) {
+      if (path.extname(file) === '.js') {
+        debug('basePath', this.configDugOptions.basePath);
+        const basePath = this.configDugOptions.basePath;
+        if (!basePath) {
+          throw new Error('basePath is not set in configDugOptions');
+        }
+        const module = await import(path.join(basePath, file));
+
+        if (typeof module === 'object') {
+          Object.assign(values, module);
+          this.recordValueOrigins(module, file);
+        }
+
+        continue;
+      }
+
+      if (path.extname(file) === '.json') {
+        const contents = await fs.readFile(file, 'utf8');
+
+        try {
+          const module = JSON.parse(contents);
+
+          if (typeof module === 'object') {
```

Should we log a warning or error if the config file isn't an object?  Might be a nicer dev experience vs. failing silently.